dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,147 @@
+            "interactions": [
+                {
+                    "name": interaction_type,
+                    "users_engaged_id": [author],
+                    "type": "receiver",
+                }
+            ],
+        }
+
+    def create_emitter_interaction(
+        self, user, period, data, interaction_type, engaged_user
+    ):
+        is_bot = self.user_bot_checker(engaged_user)
+        return {
+            "author_id": user,
+            "date": period,
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [],
+            "interactions": [
+                {
+                    "name": interaction_type,
+                    "users_engaged_id": [engaged_user],
+                    "type": "emitter",
+                }
+            ],
+        }
+
+    def create_transformed_item(self, data, period, interactions):
+        is_bot = self.user_bot_checker(data["author"])
+        return {
+            "author_id": data["author"],
+            "date": period,
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [
+                {
+                    "name": "message",
+                    "type": "emitter",
+                }
+            ],
+            "interactions": interactions,
+        }
+
+    def transform(
+        self, raw_data: list, platform_id: str, period: datetime
+    ) -> list[dict]:
+        transformed_data = []
+        for data in raw_data:
+            try:
+                if "author" not in data or not data["author"]:
+                    raise ValueError("Missing 'author' in raw data")
+
+                interactions = []
+
+                if data.get("replied_user"):
+                    interactions.append(
+                        self.create_interaction(
+                            "reply", [data["replied_user"]], "emitter"
+                        )
+                    )
+                    receiver_interaction = self.create_receiver_interaction(
+                        data, "reply", data["replied_user"], data["author"]
+                    )
+                    transformed_data.append(receiver_interaction)
+
+                if data.get("user_mentions"):
+                    interactions.append(
+                        self.create_interaction(
+                            "mention", data["user_mentions"], "emitter"
+                        )
+                    )
+                    for mentioned_user in data["user_mentions"]:
+                        mentioned_user_interaction = self.create_receiver_interaction(
```

Comment:
The author in case of the receiver of a mention would be the person that was mentioned. In this situation, we need to assign the `author=mentioned_user` and the engaged_user would be the `data["author"]` (don't forget the engaged_user we have to have a list of users)